cookiejar.go

```diff
@@ -0,0 +1,221 @@
+package fasthttp
+
+import (
+	"bytes"
+	"net"
+	"sync"
+	"time"
+)
+
+// CookieJar manages cookie storage
+type CookieJar struct {
+	m           sync.Mutex
+	hostCookies map[string][]*Cookie
+}
+
+func (cj *CookieJar) init() {
```

Comment:
I would completely remove this method and inline it in the 2 places where it's really needed.

Reply:
That way you also have less locking and unlocking as the method can just do it once instead of having it done multiple times.